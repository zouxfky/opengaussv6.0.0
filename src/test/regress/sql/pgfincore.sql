--
-- test SYSCONF
--
select true from pgsysconf();
select true from pgsysconf_pretty();

--
-- make a temp table to use below
--
CREATE TEMP TABLE test AS SELECT generate_series(1,256) as a;

--
-- this is not perfect testing but it is hard to predict what the OS will do
-- for *sure*
--

--
-- test fadvise_loader
--
select true from pgfadvise_loader('test', 'r', NULL, 0, true, true, B'1010');
select true from pgfadvise_loader('test', 'r', NULL, 0, true, false, B'1010');
select true from pgfadvise_loader('test', 'r', NULL, 0, false, true, B'1010');
select true from pgfadvise_loader('test', 'r', NULL, 0, false, false, B'1010');
-- must not fail on empty databit input
select true from pgfadvise_loader('test', 'r', NULL, 0, false, false, B'');
-- ERROR on NULL databit input
select true from pgfadvise_loader('test', 'r', NULL, 0, false, false, NULL);

--
-- test pgfincore
--
select true from pgfincore('test', true);
select true from pgfincore('test');

--
-- test DONTNEED, WILLNEED
--
select true from pgfadvise_willneed('test');
select true from pgfadvise_dontneed('test');

--
-- test PGFADVISE flags
--
select true from pgfadvise_sequential('test');
select true from pgfadvise_random('test');
select true from pgfadvise_normal('test');

--
-- tests drawers
--
select true from pgfincore('test','main',true);
