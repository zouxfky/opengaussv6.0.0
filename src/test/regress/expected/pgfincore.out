--
-- test SYSCONF
--
select true from pgsysconf();
 bool 
------
 t
(1 row)

select true from pgsysconf_pretty();
 bool 
------
 t
(1 row)

--
-- make a temp table to use below
--
CREATE TEMP TABLE test AS SELECT generate_series(1,256) as a;
--
-- this is not perfect testing but it is hard to predict what the OS will do
-- for *sure*
--
--
-- test fadvise_loader
--
select true from pgfadvise_loader('test', 'r', NULL, 0, true, true, B'1010');
 bool 
------
 t
(1 row)

select true from pgfadvise_loader('test', 'r', NULL, 0, true, false, B'1010');
 bool 
------
 t
(1 row)

select true from pgfadvise_loader('test', 'r', NULL, 0, false, true, B'1010');
 bool 
------
 t
(1 row)

select true from pgfadvise_loader('test', 'r', NULL, 0, false, false, B'1010');
 bool 
------
 t
(1 row)

-- must not fail on empty databit input
select true from pgfadvise_loader('test', 'r', NULL, 0, false, false, B'');
 bool 
------
 t
(1 row)

-- ERROR on NULL databit input
select true from pgfadvise_loader('test', 'r', NULL, 0, false, false, NULL);
ERROR:  pgfadvise_loader: databit argument shouldn't be NULL
CONTEXT:  referenced column: pgfadvise_loader
SQL function "pgfadvise_loader" statement 1
--
-- test pgfincore
--
select true from pgfincore('test', true);
 bool 
------
 t
(1 row)

select true from pgfincore('test');
 bool 
------
 t
(1 row)

--
-- test DONTNEED, WILLNEED
--
select true from pgfadvise_willneed('test');
 bool 
------
 t
(1 row)

select true from pgfadvise_dontneed('test');
 bool 
------
 t
(1 row)

--
-- test PGFADVISE flags
--
select true from pgfadvise_sequential('test');
 bool 
------
 t
(1 row)

select true from pgfadvise_random('test');
 bool 
------
 t
(1 row)

select true from pgfadvise_normal('test');
 bool 
------
 t
(1 row)

--
-- tests drawers
--
select true from pgfincore('test','main',true);
 bool 
------
 t
(1 row)

